sycl/unittests/scheduler/SchedulerTestUtils.hpp

```diff
@@ -258,7 +258,11 @@ class MockHandler : public sycl::handler {
     return CGData.MEvents;
   }
   std::vector<sycl::detail::ArgDesc> &getArgs() { return MArgs; }
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  std::string getKernelName() { return MKernelName.c_str(); }
+#else
   std::string &getKernelName() { return MKernelName; }
+#endif
```

Comment:
Why is this change needed?

Reply:
The orignal data type of MKernelName was std::string.
libsycl will call getKernelName() will return std::string which is different ABI in the SYCL header files than libsycl.
So, returning std::string directly is not going to work.
We need to marshall the std::string into detail::string and reconstruct it in libsycl.

Reply:
My bad. True. Thanks for explaining!